emscripten.py

```diff
@@ -1282,10 +1282,12 @@ def create_exports(exported_implemented_functions, in_table, function_table_data
   quote = quoter()
   asm_runtime_funcs = create_asm_runtime_funcs()
   all_exported = exported_implemented_functions + asm_runtime_funcs + function_tables(function_table_data)
-  # in a wasm shared library + emulated function pointers, export all the table so that
-  # we can easily find the function pointer for each function
-  if (shared.Settings.RELOCATABLE or not shared.Settings.WASM) and \
-     shared.Settings.EMULATED_FUNCTION_POINTERS:
+  # In an asm shared library + emulated function pointers, export all the table so that
+  # we can easily find the function pointer for each function. The reason is that for asm, we use 
+  # JS to add the side module's functions to the shared table with the parent, so they must 
+  # be exported for that JS to access them. In wasm we don't need that, as the wasm can 
+  # directly add its functions to the Table. 
+  if (not shared.Settings.WASM and shared.Settings.EMULATED_FUNCTION_POINTERS):
```

Comment:
parentheses are not necessary here

Reply:
oh, and also, let's not regress the asm.js case: add the `and shared.Settings.RELOCATABLE` stuff back please

Reply:
Sure @kripken , I am already in the AUTHORS list. 

Just to clarify, shared.Settings.RELOCATABLE is set by default for dynamic linking, so the check will be like this?

if (shared.Settings.RELOCATABLE and not shared.Settings.WASM) and shared.Settings.EMULATED_FUNCTION_POINTERS):

Reply:
Yes, but without any of the `(` and `)`

Reply:
Oh ok, but the original code had it though e.g.

if (shared.Settings.RELOCATABLE or not shared.Settings.WASM) and shared.Settings.EMULATED_FUNCTION_POINTERS:

Reply:
Yes, but that's because the original had `(X or Y) and Z`, while now we have `A and B and C`. With just `and`s, there is no real need for parentheses, and I don't think they help clarity here either, so may as well remove them.

Reply:
Thanks for the clarification @kripken :) Unfortunately, I added shared.Settings.RELOCATABLE to the check and the following test failed. It looks like we need to export the internal functions for asm even when RELOCATABLE is not set? This corresponds with the current behaviour in production actually.

===============================================================
ERROR: test_add_function (test_core.asmi)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\programs\emsdk\python\2.7.13.1_64bit\python-2.7.13.amd64\lib\unittest\case.py", line 329, in run
    testMethod()
  File "D:\programs\emsdk\emscripten\incoming\tests\test_core.py", line 54, in decorated
    f(self)
  File "D:\programs\emsdk\emscripten\incoming\tests\test_core.py", line 6687, in test_add_function
    self.do_run_from_file(src, expected)
  File "D:\programs\emsdk\emscripten\incoming\tests\runner.py", line 758, in do_run_from_file
    self.do_run(open(src).read(), open(expected_output).read(), *args, **kwargs)
  File "D:\programs\emsdk\emscripten\incoming\tests\runner.py", line 801, in do_run
    raise e
Exception: Expected to find 'Hello 7 from JS!
ok
' in 'exception thrown: TypeError: FUNCTION_TABLE_iiii[x] is not a function,TypeError: FUNCTION_TABLE_iiii[x] is not a function

Reply:
Anyway, I have updated the PR by removing the unnecessary parentheses. Let me know if the shared.Settings.RELOCATABLE check is still an issue then. Thanks @kripken.